### PR TITLE
k8s: Use ParseService when comparing two services

### LINKED ISSUE #8700: Changing Service type or ports does not trigger ServiceUpdate event
To reproduce:

1. Create `ClusterIP` service `foo`.
2. Patch `foo` with `kubectl edit svc foo` to change its type to `NodePort`.
3. No `Kubernetes service definition changed` log entries (level=debug) = no updates in the LB BPF maps to reflect the change.

### PR DESCRIPTION
**Pull request by brb**
Previously, the `EqualV1Services()` function had its own `k8s.Service` constructor which was not in sync with `ParseService()` and did not consider some service fields.

The consequence of this was that changing a type or ports of a service did not trigger the service update, and thus the service change was not reflected in the SVC BPF maps.

Use `ParseService()` instead of the custom constructor to prevent any discrepancies in the future.

Fix #8700.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9689)

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

File: pkg/k8s/factory_functions.go [hunk 1]
```diff
@@ -126,19 +124,8 @@ func EqualV1Services(svc1, svc2 *types.Service) bool {
...
-	clusterIP = net.ParseIP(svc2.Spec.ClusterIP)
-	headless = false
-	if strings.ToLower(svc2.Spec.ClusterIP) == "none" {
-		headless = true
-	}
-	si2 := NewService(clusterIP, svc2.Spec.ExternalIPs, headless, svc2.Labels, svc2.Spec.Selector)
+	_, si1 := ParseService(svc1)
+	_, si2 := ParseService(svc2)
 
 	// Please write all the equalness logic inside the K8sServiceInfo.Equals()
```
Comment (aanm) [inline pkg/k8s/factory_functions.go:134]: Please add `if sid1 != sid2 { return false }`

File: pkg/k8s/factory_functions.go [hunk 2]
```diff
@@ -126,19 +124,8 @@ func EqualV1Services(svc1, svc2 *types.Service) bool {
...
-	si1 := NewService(clusterIP, svc1.Spec.ExternalIPs, headless, svc1.Labels, svc1.Spec.Selector)
-
-	clusterIP = net.ParseIP(svc2.Spec.ClusterIP)
-	headless = false
-	if strings.ToLower(svc2.Spec.ClusterIP) == "none" {
-		headless = true
-	}
-	si2 := NewService(clusterIP, svc2.Spec.ExternalIPs, headless, svc2.Labels, svc2.Spec.Selector)
+	_, si1 := ParseService(svc1)
+	_, si2 := ParseService(svc2)
```
Comment (aanm) [inline pkg/k8s/factory_functions.go]:
```
sid1, si1:=...
sid2, si2:=...
```

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/k8s/factory_functions.go (hunk 1 above)
Comment (brb) [reply pkg/k8s/factory_functions.go:134]: Previously, it didn't check the IDs, but OK, I will add.

**Comment by brb** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Pull request merged**